src/library/scala/collection/immutable/Vector.scala

```diff
@@ -88,7 +83,7 @@ override def companion: GenericCompanion[Vector] = Vector
 
   override def lengthCompare(len: Int): Int = length - len
 
-  private[collection] final def initIterator[B >: A](s: VectorIterator[B]) {
+  private[collection] def initIterator[B >: A](s: VectorIterator[B]) {
```

Comment:
If we can get away with finality, we should leave it in.  In general, immutable data structures do not lend themselves well to subclassing.